networkit/cpp/planarity/LeftRightPlanarityCheck.cpp

```diff
@@ -0,0 +1,262 @@
+        }
+        if (tmpConflictPair.left.isEmpty()) {
+            tmpConflictPair.left = currentConflictPair.left;
+        } else {
+            ref[tmpConflictPair.left.low] = currentConflictPair.left.high;
+        }
+        tmpConflictPair.left.low = currentConflictPair.left.low;
+    }
+
+    if (!tmpConflictPair.left.isEmpty() || !tmpConflictPair.right.isEmpty())
+        stack.push(tmpConflictPair);
+    return true;
+}
+
+count LeftRightPlanarityCheck::getLowestLowPoint(const ConflictPair &conflictPair) {
+    if (conflictPair.left.isEmpty())
+        return lowestPoint[conflictPair.right.low];
+    if (conflictPair.right.isEmpty())
+        return lowestPoint[conflictPair.left.low];
+    return std::min(lowestPoint[conflictPair.right.low], lowestPoint[conflictPair.left.low]);
+}
+
+void LeftRightPlanarityCheck::removeBackEdges(Edge edge) {
+    auto parentNode = edge.u;
+    while (!stack.empty() && getLowestLowPoint(stack.top()) == heights[parentNode]) {
+        stack.pop();
+    }
+
+    if (!stack.empty()) {
+        auto conflictPair = stack.top();
+        stack.pop();
+        while (conflictPair.left.high != noneEdge && conflictPair.left.high.v == parentNode) {
+            auto it = ref.find(conflictPair.left.high);
+            conflictPair.left.high = (it != ref.end()) ? it->second : noneEdge;
+        }
+        if (conflictPair.left.high == noneEdge && conflictPair.left.low != noneEdge) {
+            ref[conflictPair.left.low] = conflictPair.right.low;
+            conflictPair.left.low = noneEdge;
+        }
+        while (conflictPair.right.high != noneEdge && conflictPair.right.high.v == parentNode) {
+            auto it = ref.find(conflictPair.right.high);
+            conflictPair.right.high = (it != ref.end()) ? it->second : noneEdge;
+        }
+
+        if (conflictPair.right.high == noneEdge && conflictPair.right.low != noneEdge) {
+            ref[conflictPair.right.low] = conflictPair.left.low;
+            conflictPair.right.low = noneEdge;
+        }
+        stack.push(conflictPair);
+    }
+
+    if (!stack.empty() && lowestPoint[edge] < heights[parentNode]) {
+        auto highestReturnEdgeLeft = stack.top().left.high;
+        auto highestReturnEdgeRight = stack.top().right.high;
+        if (highestReturnEdgeLeft != noneEdge
+            && (highestReturnEdgeRight != noneEdge
+                || lowestPoint[highestReturnEdgeLeft] > lowestPoint[highestReturnEdgeRight])) {
+            ref[edge] = highestReturnEdgeLeft;
+        } else {
+            ref[edge] = highestReturnEdgeRight;
+        }
+    }
+}
+
+bool LeftRightPlanarityCheck::dfsTesting(node startNode) {
+    std::stack<node> dfsStack;
+    dfsStack.emplace(startNode);
+    auto neighborIterators =
+        std::unordered_map<node, decltype(dfsGraph.neighborRange(startNode).begin())>{};
+
+    auto preprocessedEdges = std::unordered_set<Edge>{};
+    while (!dfsStack.empty()) {
```

Comment:
Nit: `dfsStack` is never empty at the beginning, so you can replace this with a `do {} while (!dfsStack.empty());` loop.